l1-contracts/contracts/chain-registrar/ChainRegistrar.sol

```diff
@@ -0,0 +1,196 @@
+        /// @notice Address of the main contract (diamond proxy) for the deployed chain.
+        address diamondProxy;
+        /// @notice Address of the L2 bridge inside the deployed chain.
+        address l2BridgeAddress;
+    }
+
+    /// @notice Initializes the contract with the given parameters.
+    /// @dev Can only be called once, during contract deployment.
+    /// @param _bridgehub Address of the ZKsync Bridgehub.
+    /// @param _l2Deployer Address of the L2 deployer.
+    /// @param _owner Address of the contract owner.
+    function initialize(address _bridgehub, address _l2Deployer, address _owner) public {
+        bridgehub = IBridgehub(_bridgehub);
+        l2Deployer = _l2Deployer;
+        _transferOwnership(_owner);
+    }
+
+    /// @notice Proposes a new chain to be registered in the zkSync ecosystem.
+    /// @dev The proposal will fail if the chain has already been registered.
+    /// @dev For non-ETH-based chains, either an equivalent of 1 ETH of the base token must be approved or transferred to the L2 deployer.
+    /// @param _chainId Unique ID of the proposed chain.
+    /// @param _pubdataPricingMode Mode for charging users for pubdata.
+    /// @param _blobOperator Address responsible for commit transactions.
+    /// @param _operator Address responsible for prove and execute transactions.
+    /// @param _governor Address to receive ownership of the ChainAdmin contract.
+    /// @param _baseTokenAddress Address of the base token used for gas fees.
+    /// @param _tokenMultiplierSetter Address responsible for setting the base token multiplier.
+    /// @param _gasPriceMultiplierNominator Gas price multiplier numerator for L1->L2 transactions.
+    /// @param _gasPriceMultiplierDenominator Gas price multiplier denominator for L1->L2 transactions.
+    function proposeChainRegistration(
+        uint256 _chainId,
+        PubdataPricingMode _pubdataPricingMode,
+        address _blobOperator,
+        address _operator,
+        address _governor,
+        address _baseTokenAddress,
+        address _tokenMultiplierSetter,
+        uint128 _gasPriceMultiplierNominator,
+        uint128 _gasPriceMultiplierDenominator
+    ) external {
+        ChainConfig memory config = ChainConfig({
+            chainId: _chainId,
+            pubdataPricingMode: _pubdataPricingMode,
+            blobOperator: _blobOperator,
+            operator: _operator,
+            governor: _governor,
+            baseToken: BaseToken({
+                tokenAddress: _baseTokenAddress,
+                tokenMultiplierSetter: _tokenMultiplierSetter,
+                gasPriceMultiplierNominator: _gasPriceMultiplierNominator,
+                gasPriceMultiplierDenominator: _gasPriceMultiplierDenominator
+            })
+        });
+
+        if (bridgehub.stateTransitionManager(config.chainId) != address(0)) {
+            revert ChainIsAlreadyDeployed();
+        }
+
+        proposedChains[msg.sender][_chainId] = config;
```

Comment:
Do you think it makes sense to allow user to re-propose config? I would rather restrict it to make avoid any confusion on the final config (as soon as config is onchain it immutable).

Reply:
@Deniallugo this should be fixed too

Reply:
<img width="664" alt="image" src="https://github.com/user-attachments/assets/36906b26-99a7-4933-a552-a16ebb52e9e4">
It's done 3 lines before